common/analysis/processingChainController.js

```diff
@@ -0,0 +1,43 @@
+import * as Chain from './processingChain'
+import * as Step from './processingStep'
+import * as StepVariable from './processingStepVariable'
+import * as Calculation from './processingStepCalculation'
+
+export const assocCalculation = ({ chain, step, calculation }) => {
+  const stepUpdated = Step.assocCalculation(calculation)(step)
```

Comment:
all update methods should belong here now. move that inside or inner method

Reply:
done